### PR TITLE
Clean up KFP v2 pipeline documentation: components

### PR DESCRIPTION
**Pull request by droctothorpe**

### CODE REVIEW

File: content/en/docs/components/pipelines/v2/components/container-components.md [hunk 1]
```diff
@@ -67,8 +68,8 @@ def say_hello(name: str):
 When you run the component with the argument `name='World'`, you’ll see the string `'Hello, World!'` in `say_hello`’s logs.
 
 ### Create component outputs
- 
-Unlike Python functions, containers do not have a standard mechanism for returning values. To enable Container Components to have outputs, KFP requires you to write outputs to a file inside the container. KFP will read this file and persist the output to [ML Metadata][ml-metadata].
+
+Unlike Python functions, containers do not have a standard mechanism for returning values. To enable Container Components to have outputs, KFP requires you to write outputs to a file inside the container. KFP will read this file, persist the output to your [Artifact][artifacts] store, and register it in [ML Metadata][ml-metadata].
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/components/container-components.md:72]: Good thinking. This should be clearer.

For parameters, I think the current explanation is correct: KFP reads the JSON from the file and persists the output parameter values to ML Metadata. For artifacts, the proposed is correct: KFP copies the artifact from the path local path (`Artifact().path`) to the cloud object store. For artifact, KFP also reads the artifact metadata from `executor_output.json` and updates the metadata in ML Metadata. (The treatment of artifact metadata in `executor_output.json` is not discussed at all in this section.)

This is one of those cases where it's tough to fully describe container components without fully describing artifacts, the object storage / pipeline root, volume mounting / copying of artifact files, the idea of metadata, and the schema for the `executor_output.json` file (currently not addressed anywhere either)...

Any thoughts on how to proceed?

**Review by connor-mccarthy — COMMENTED** [review]
One comment/question, @droctothorpe! Thanks again for your attention to detail!

File: content/en/docs/components/pipelines/v2/components/container-components.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (droctothorpe) [reply content/en/docs/components/pipelines/v2/components/container-components.md:72]: Thanks for the clarification! Agreed that that's quite a rabbit hole.

What do you think about just a simple:

```KFP will read this file and persist the output.```

Alongside providing a more in-depth treatment in a separate document like you suggested (something like `Component IO`).

[the remark this answers is not in the document]
Comment (droctothorpe) [reply content/en/docs/components/pipelines/v2/components/container-components.md:72]: More than happy to revert to the original wording as well if that's clearer in the short term!

File: content/en/docs/components/pipelines/v2/components/container-components.md [hunk 2]
```diff
@@ -123,4 +125,4 @@ This should look very similar to the [Hello World pipeline][hello-world-pipeline
 [dsl-outputpath]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/dsl.html#kfp.dsl.OutputPath
 [dsl-container-component]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/dsl.html#kfp.dsl.container_component
 [dsl-containerspec]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/dsl.html#kfp.dsl.ContainerSpec
-[ml-metadata]: https://github.com/google/ml-metadata
\ No newline at end of file
+[ml-metadata]: https://github.com/google/ml-metadata
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/components/container-components.md:128]: Can we remove this link now?

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]
One nit, then let's merge!

File: content/en/docs/components/pipelines/v2/components/container-components.md (hunk 2 above)
Comment (droctothorpe) [reply content/en/docs/components/pipelines/v2/components/container-components.md]: Done! 👍

**Comment by connor-mccarthy** [comment]
/lgtm
/approve

Thank you again, @droctothorpe!

**Pull request merged**